<a href="https://colab.research.google.com/github/chanikyavardhan/fml/blob/main/Experiment_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report,confusion_matrix 

In [5]:
data=pd.read_csv("diabetes.csv")
data.head()

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0            6      148         72       35        0  33.6  0.627   50   
1            1       85         66       29        0  26.6  0.351   31   
2            8      183         64        0        0  23.3  0.672   32   
3            1       89         66       23       94  28.1  0.167   21   
4            0      137         40       35      168  43.1  2.288   33   

   diabetes  
0         1  
1         0  
2         1  
3         0  
4         1

In [7]:
x=data.drop('diabetes',axis=1).values
y=data['diabetes'].values

In [8]:
#split the dataset into training and testing data
X_train,X_test,y_train,y_test=train_test_split(x,y,test_size=0.1,random_state=42)

In [9]:
#Normalize the data
scaler = StandardScaler()
X_train=scaler.fit_transform(X_train)

In [10]:
#fitting the model
model=LogisticRegression(C=0.01,random_state=42)
model.fit(X_train,y_train)

LogisticRegression(C=0.01, random_state=42)

In [15]:
#Normalizing of testing data
X_test = scaler.transform(X_test)
y_pred = model.predict(X_test)
model.score(X_train,y_train)

0.7713458755426917

In [16]:
model.score(X_test,y_test)

0.6493506493506493

In [17]:
#Confusion matrix for evaluation of model
confusion_matrix(y_test,y_pred)


array([[44,  6],
       [21,  6]])

In [18]:
print(classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.68      0.88      0.77        50
           1       0.50      0.22      0.31        27

    accuracy                           0.65        77
   macro avg       0.59      0.55      0.54        77
weighted avg       0.61      0.65      0.60        77



**Conclusion:**The accuracy of the training data is 0.65 and for the testing data is 0.60